### ISSUE TITLE
I am confused about the initElapsedS & readElapsedS.

### ISSUE THREAD

**Issue by CKopoer** [UoB-HPC/BabelStream#173]
So, As these [lines](https://github.com/UoB-HPC/BabelStream/blob/main/src/main.cpp#L342) shows, we get the initElapsedS & readElapsedS from the difference of each other. Is't a mistake or something meaningful I haven't understood?
![image](https://github.com/UoB-HPC/BabelStream/assets/52709412/ed574d64-f158-46fb-8745-a91edb075445)

Otherwise, I get the results on H800 using another closed-source NV-STREAM tool. It seems that it provided better bandwidth performance result compared with BabelStream because of the optimized block size parameters. What's more, it also  and show Read & Write results. Could I take the Init_kernel as the Write result and read_arrays as Read result in BabelStream?
<img width="657" alt="image" src="https://github.com/UoB-HPC/BabelStream/assets/52709412/4715b444-ccc7-4aca-b929-75caf41eff29">
<img width="611" alt="image" src="https://github.com/UoB-HPC/BabelStream/assets/52709412/360c16b2-6918-4ce0-a94d-7d52ed9f0dd4">

**Comment by tom91136** [comment]
Yes, Init and Read is a new thing we report that measures the setup and read-back time of the buffers.
BabelStream does not have an direct equivalent of the Read and Write kernel in NV-STREAM.
> It seems that it provided better bandwidth performance result compared with BabelStream because of the optimized block size parameters.

What's the performance difference you're observing?

**Comment by gonzalobg** [comment]
@CKopoer the time intervals used for `Init` and `Read` are incorrect. 
#186 fixes it (among other things).

> Could I take the Init_kernel as the Write result and read_arrays as Read result in BabelStream?

The `Init` and `Read` timings stem from a single measurement, and are not intended to be a measure of `Read` and `Write` bandwidth.

After #186 is merged, adding proper `Read` & `Write` bandwidth benchmarks is straightforward (although still quite a bit of work since they need to be added to all languages). I agree that these two help paint a more complete picture about the hardware than `Copy`.

**Comment by gonzalobg** [comment]
Fixed by #186 .
